src/widgets/list.rs

```diff
@@ -186,35 +186,82 @@ impl<'a> List<'a> {
         offset: usize,
         max_height: usize,
     ) -> (usize, usize) {
+        // In case the offset is greater than the number of items, set the offset
+        // to be the index of the last item. Otherwise, use the provided offset
         let offset = offset.min(self.items.len().saturating_sub(1));
+
+        // Index of the first list item we're rendering
         let mut start = offset;
+
+        // Index of the last item item we're able to render in the given area
         let mut end = offset;
```

Comment:
I'm not a big fan of documenting everything and I think some of the variables/logic (e.g. this one) explains itself. I would like to get the opinions of other maintainers (cc @ratatui-org/core) but I would prefer if we have a doc/function comment explaining the general purpose/flow of the function rather than having distinct comments throughout the function (unless they are really necessary).

Reply:
I agree not all of the comments added are necessary and pieces like what you are highlighting here could absolutely be consolidated at a minimum. Leaving comments was helpful with trying to fully understand how lists are rendered (considering there was nothing prior to this) and I figured someone else may find it useful if changes need to be made down the line.

Agreed that a doc/function comment should be added regardless - would be happy to add that.